### ISSUE TITLE
Suggest using is_ascii_lowercase / is_ascii_uppercase

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#9290]
### What it does

This simple lint suggests to replace matches!() on full upper/lower case ranges with the built-in methods.

### Lint Name

use_is_ascii_lower_upper_case

### Category

style

### Advantage

The code is simpler, it contains less tokens.

### Drawbacks

None, I think.

### Example

```rust
#![warn(clippy::all)]
#![warn(clippy::nursery)]
#![warn(clippy::pedantic)]
fn main() {
    assert!(matches!('x', 'a' ..= 'z'));
    assert!(matches!('X', 'A' ..= 'Z'));
    assert!(matches!(b'x', b'a' ..= b'z'));
    assert!(matches!(b'X', b'A' ..= b'Z'));
}
```

Could be written as:

```rust
#![warn(clippy::all)]
#![warn(clippy::nursery)]
#![warn(clippy::pedantic)]
fn main() {
    assert!('x'.is_ascii_lowercase());
    assert!(b'x'.is_ascii_lowercase());
    assert!('X'.is_ascii_uppercase());
    assert!(b'X'.is_ascii_uppercase());    
}
```

**Comment by nerdypepper** [comment]
this should be applicable to `'a'..='z'.contains(&c)` as well.

**Comment by koka831** [comment]
It might be better to use `matches!` macro when given pattern contains multiple conditions.

```rust
assert!(matches!(foo, 'A'..='Z' | 'a'..='z'));

// should be?
assert!(foo.is_ascii_uppercase() || foo.is_ascii_lowercase())
```

Of course it'd be great to suggest to use `is_ascii_alphabetic` in that case.

**Comment by xFrednet** [comment]
> It might be better to use `matches!` macro when given pattern contains multiple conditions.

For the given example, suggesting `is_ascii_alphabetic` would be ideal, but also a lot more work to implement. So my suggestion would be to first focus on `is_ascii_uppercase` and `is_ascii_lowercase`. The lint can always be improved later :)

Though, I would try to find a more general lint name like `manual_is_ascii_check`. Then we can also use that for [`is_ascii_digit`](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_digit). The lint description could also say that these `is_ascii_` functions are often const :)

**Comment by koka831** [comment]
> The lint can always be improved later :)

Yes, that's true. we'll start small and then grow it.
So I'll try to handle `matches!` macro with single range as suggested above.

**Comment by xFrednet** [comment]
Oh yes, we can close this. Thank you for the ping and the implementation! :)
